iOS_SDK/OneSignalSDK/Source/Requests.m

```diff
@@ -143,14 +149,15 @@ + (instancetype _Nonnull)withUserId:(NSString * _Nonnull)userId appId:(NSString
 @end
 
 @implementation OSRequestCreateDevice
-+ (instancetype _Nonnull)withAppId:(NSString * _Nonnull)appId withDeviceType:(NSNumber * _Nonnull)deviceType withEmail:(NSString * _Nullable)email withPlayerId:(NSString * _Nullable)playerId withEmailAuthHash:(NSString * _Nullable)emailAuthHash {
++ (instancetype _Nonnull)withAppId:(NSString * _Nonnull)appId withDeviceType:(NSNumber * _Nonnull)deviceType withEmail:(NSString * _Nullable)email withPlayerId:(NSString * _Nullable)playerId withEmailAuthHash:(NSString * _Nullable)emailAuthHash withExternalIdAuthToken:(NSString * _Nullable)externalIdAuthToken {
     let request = [OSRequestCreateDevice new];
     
     request.parameters = @{
        @"app_id" : appId,
        @"device_type" : deviceType,
        @"identifier" : email ?: [NSNull null],
        @"email_auth_hash" : emailAuthHash ?: [NSNull null],
+       @"external_user_id_auth_hash" : externalIdAuthToken ?: [NSNull null],
```

Comment:
Not sure why this is showing up as an addition. This `external_user_id_auth_hash` line was already present in the 2.16.1 release
https://github.com/OneSignal/OneSignal-iOS-SDK/blob/2.16.1/iOS_SDK/OneSignalSDK/Source/Requests.m#L160

Reply:
The 2.16.1 release branch was deleted and I was not able to recover it so this is based off of the 2.16.0 release branch with the 2.16.1 commits added to it. If there is a way to recover that branch I could rebase to it

Reply:
ah, there is still a `2.16.1` tag which should point to a specific commit on master which might work. If that isn't easy todo I reviewed this PR by using the following view and it looks good to me.
https://github.com/OneSignal/OneSignal-iOS-SDK/compare/2.16.1..sdk_2.X.X